javascript/javascript3/week3/TASK3-Shopping cart/index.js

```diff
@@ -1,130 +1,149 @@
+      fragment.appendChild(ul);
+      this.products.forEach((product) => {
+        createListElement(product);
+      });
+
+      function createListElement(productToList) {
+        let mainLI = document.createElement("LI");
+        ul.appendChild(mainLI);
+        let productUL = document.createElement("UL");
+        mainLI.appendChild(productUL);
+        let nameLI = document.createElement("LI");
+        nameLI.textContent = `Procuct: ${productToList.name}`;
+        productUL.append(nameLI);
+        let priceLI = document.createElement("LI");
+        priceLI.textContent = `Price: ${productToList.price}`;
+        productUL.append(priceLI);
+        document.body.appendChild(fragment);
+      }
+    }
 
-  renderProducts() {
-    const ul = document.createElement("UL");
-    document.body.appendChild(ul);
-    this.products.forEach((product) => {
-      let mainLI = document.createElement("LI");
-      ul.appendChild(mainLI);
-      let productUL = document.createElement("UL");
-      mainLI.appendChild(productUL);
-      let nameLI = document.createElement("LI");
-      nameLI.innerHTML = `Procuct: ${product.name}`;
-      productUL.append(nameLI);
-      let priceLI = document.createElement("LI");
-      priceLI.innerHTML = `Price: ${product.price}`;
-      productUL.append(priceLI);
-    });
+    getUser = async (userID) => {
+      try {
+        const baseUrl = "https://jsonplaceholder.typicode.com/users/";
+        const query = userID;
+        return await getData(baseUrl, query);
+      } catch (error) {
+        console.log(error);
+      }
+    };
   }
 
-  getUser = async () => {
+  getData = async (baseUrl, query) => {
     try {
-      let url = `https://jsonplaceholder.typicode.com/users/1`;
-      const response = await fetch(url);
+      const response = await fetch(`${baseUrl}${query}`);
       if (response.status >= 200 && response.status < 300) {
-        let user = response;
-        user = user.json();
-        return user;
+        const data = response.json();
+        return data;
       } else {
         throw new Error(`HTTP error! status: ${response.status}`);
       }
     } catch (error) {
       console.log(error);
     }
   };
-}
-
-const shoppingCart = new ShoppingCart();
 
-const flatscreen = new Product("flat-screen", 1000);
-const tablet = new Product("tablet", 2000);
-const notebook = new Product("notebook", 3000);
-const mobilephone = new Product("mobilephone", 4000);
-const pc = new Product("pc", 5000);
-
-shoppingCart.addProduct(flatscreen);
-shoppingCart.addProduct(tablet);
-shoppingCart.addProduct(notebook);
-shoppingCart.addProduct(mobilephone);
-shoppingCart.addProduct(pc);
-
-shoppingCart.removeProduct(flatscreen);
-
-console.log(shoppingCart.getTotal());
-console.log(shoppingCart.searchProduct("tablet"));
+  async function getCurrencyRate(amountToConvert, convertToCurrency) {
+    const baseUrl = `https://v6.exchangerate-api.com/v6/84a875e2ff2c28d158651393/latest/`;
```

Comment:
again, take this out of the function and for naming convention i would use `BASE_URL` to signify a constant